cmd/state/internal/cmdtree/ppm.go

```diff
@@ -1,135 +1,134 @@
 package cmdtree
 
 import (
-	"fmt"
-	"os"
-	"os/exec"
 	"strings"
 
-	"github.com/ActiveState/cli/internal/analytics"
 	"github.com/ActiveState/cli/internal/captain"
-	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/internal/primer"
-	"github.com/ActiveState/cli/pkg/projectfile"
+	"github.com/ActiveState/cli/internal/runners/ppm"
 )
 
-func printSuggestion(prime *primer.Values, ppmIntent, newCommand, docLink string) error {
-	prime.Output().Print(locale.Tr("ppm_print_suggestion", ppmIntent, newCommand, docLink))
+func printSuggestion(out output.Outputer, ppmIntent, newCommand, docLink string) error {
+	out.Print(locale.Tr("ppm_print_suggestion", ppmIntent, newCommand, docLink))
 	return nil
 }
 
-func printDefault(prime *primer.Values) error {
-	prime.Output().Print(strings.TrimSpace(locale.T("ppm_header_message")))
+func printDefault(out output.Outputer) error {
+	out.Print(strings.TrimSpace(locale.T("ppm_header_message")))
 	return nil
 }
 
-func printMain(prime *primer.Values) error {
-	prime.Output().Print(locale.T("ppm_print_main"))
+func printMain(out output.Outputer) error {
+	out.Print(locale.T("ppm_print_main"))
 	return nil
 }
 
 func newPpmCommand(prime *primer.Values) *captain.Command {
+	shim := ppm.NewShim(prime.Output())
 	rootCmd := captain.NewHiddenShimCommand(
 		"_ppm",
 		nil, nil,
 		func(_ *captain.Command, args []string) error {
 			for _, arg := range args {
 				if arg == "--version" {
-					return printDefault(prime)
+					return printDefault(prime.Output())
 				}
 			}
-			return shim(prime, "ppm", "packages", "ppm_print_forward", args...)
+			return shim.RunPPM(args...)
 		},
 	)
 
 	var children []*captain.Command
-	children = addPackagesCommands(prime, children)
-	children = addRepositoryCommands(prime, children)
-	children = addProjectCommands(prime, children)
-	children = addVersionCommand(prime, children)
-	children = addInfoCommand(prime, children)
-	children = addOtherCommands(prime, children)
+	children = addPackagesCommands(prime.Output(), children)
+	children = addRepositoryCommands(prime.Output(), children)
+	children = addProjectCommands(prime.Output(), children)
+	children = addVersionCommand(prime.Output(), children)
+	children = addInfoCommand(prime.Output(), children)
+	children = addOtherCommands(prime.Output(), children)
```

Comment:
I was only flagging this for your print functions, here it still makes sense to pass the whole prime structure as that's the convention.